From 492ed3575e7e7a5878572781937cbab04f27c2d7 Mon Sep 17 00:00:00 2001
From: Orion Hodson <oth@google.com>
Date: Fri, 28 Jun 2019 11:38:02 +0100
Subject: [PATCH 1/7] JIT: fix executable flush range

Flushing in executable memory should exactly mirror range in
non-executable memory.

Test: Manual (see bug)
Bug: 132205399
Merged-In: I19d2e8e69eef07648803937fff928aa9b45ac5a2
Change-Id: I6a26e4aa236e190ce4117c0aa20f6067892d7f1c
---
 runtime/jit/jit_code_cache.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/jit/jit_code_cache.cc b/runtime/jit/jit_code_cache.cc
index 333f2da136..a6aefc4773 100644
--- a/runtime/jit/jit_code_cache.cc
+++ b/runtime/jit/jit_code_cache.cc
@@ -1060,8 +1060,8 @@ uint8_t* JitCodeCache::CommitCodeInternal(Thread* self,
     }
 
     // Invalidate i-cache for the executable mapping.
-    uint8_t* x_memory = reinterpret_cast<uint8_t*>(method_header);
     if (cache_flush_success) {
+      uint8_t* x_memory = reinterpret_cast<uint8_t*>(FromCodeToAllocation(code_ptr));
       cache_flush_success = FlushCpuCaches(x_memory, x_memory + total_size);
     }
 
-- 
2.34.1

